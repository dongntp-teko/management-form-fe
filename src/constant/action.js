// Created by thanhpd on 6/11/2019
// @flow

export default {
  CREATE: 'create',
  UPDATE: 'update',
  READ: 'read',
  DELETE: 'delete',
}
