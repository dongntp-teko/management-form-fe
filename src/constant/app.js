// Created by thanhpd on 6/11/2019
// @flow

export default {
  INTERN: 'intern',
}
