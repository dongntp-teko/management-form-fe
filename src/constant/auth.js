// Created by thanhpd 2/18/2019
// @flow

export default ({
  KEY_CURRENT_PERMISSIONS: 'currentPermissions',
  KEY_CURRENT_USER: 'currentUser',
  api: {
    USER_DETAIL: '/me2',
  },
}: {|
  api: Object,
  KEY_CURRENT_PERMISSIONS: string,
  KEY_CURRENT_USER: string,
|});
