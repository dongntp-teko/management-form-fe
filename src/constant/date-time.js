// Created by thanhpd on 6/17/2019

const DATE_FORMAT = 'DD/MM/YYYY';
const TIME_FORMAT = 'HH:mm:ss';
const TIME_WITHOUT_SECOND = 'HH:mm';
const DATE_TIME_FORMAT = 'DD/MM/YYYY HH:mm:ss';
const HOUR_MINUTE_FORMAT = 'HH:mm';
const DATE_HOUR_MINUTE_FORMAT = 'DD/MM/YYYY HH:mm';

export default {
  DATE_FORMAT,
  TIME_FORMAT,
  TIME_WITHOUT_SECOND,
  DATE_TIME_FORMAT,
  HOUR_MINUTE_FORMAT,
  DATE_HOUR_MINUTE_FORMAT,
}
