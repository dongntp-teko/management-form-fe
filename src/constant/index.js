export { default as localStorageConstants } from 'constant/local-storage';
export { default as authConstants } from 'constant/auth';
export { default as paginationConstants } from 'constant/pagination';
export { default as notificationConstants } from 'constant/notification';
export { default as appConstants } from 'constant/app';
export { default as resourceConstants } from 'constant/resource';
export { default as actionConstants } from 'constant/action';
export { default as defaultConfigs } from 'constant/defaultConfigs';
export { default as languageConstants } from 'constant/language';
export { default as dateTimeConstants } from 'constant/date-time';
