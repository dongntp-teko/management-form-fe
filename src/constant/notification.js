// Created by thanhpd on 5/24/2019
// @flow

const types = {
  danger: {
    text: 'Danger',
    value: 'danger',
  },
  warning: {
    text: 'Warning',
    value: 'warning',
  },
  success: {
    text: 'Success',
    value: 'success',
  },
  info: {
    text: 'Info',
    value: 'info',
  },
};

export default {
  types,
}
