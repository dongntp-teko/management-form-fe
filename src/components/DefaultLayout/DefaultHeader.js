// @flow
import React, { useEffect, useState } from 'react';
import { Badge, DropdownItem, DropdownMenu, DropdownToggle, Nav, NavItem, NavLink } from 'reactstrap';

import { AppHeaderDropdown, AppNavbarBrand, AppSidebarToggler } from '@coreui/react';
import logo from 'assets/img/brand/logoPV.svg';
import { useGlobal } from 'reactn';
import { dateTimeHelper, notificationHelper } from 'helpers';
import moment from 'moment';
import _ from 'lodash';
import { authConstants, localStorageConstants } from 'constant';
import { regions } from 'localization';
import Region from 'components/Region';
import NotificationModal from './NotificationsModal';

const originalSetItem = localStorage.setItem;
const KEY_EVENT_INSERTED = 'itemInserted';
const dropDownStyle = { right: 'auto', width: 400 };

// eslint-disable-next-line func-names

const setItemDescriptor = Object.getOwnPropertyDescriptor(
  localStorage,
  'setItem',
);
if (!setItemDescriptor || setItemDescriptor.writeable) {
  localStorage.setItem = function() {
    const event = new Event(KEY_EVENT_INSERTED);
    // eslint-disable-next-line prefer-rest-params
    originalSetItem.apply(this, arguments);

    document.dispatchEvent(event);
  };
}

const readCurrentNotifications = setNotifications => {
  const notifications = notificationHelper.getNotificationsForCurrentUser();
  setNotifications(notifications);
};

export default (props: Object) => {
  const [currentUser] = useGlobal(authConstants.KEY_CURRENT_USER);
  const [modalVisible, setModalVisible] = useState(false);

  const [notifications, setNotifications] = useGlobal(
    localStorageConstants.NOTIFICATION_KEY,
  );
  const sortedNotifications = _.orderBy(notifications, ['createdAt'], ['desc']);
  const totalUnSeenNotification = sortedNotifications.filter(n => !n.seen)
    .length;
  const seenAllNotifications = () => {
    const seenNotifications = notifications.map(n => ({ ...n, seen: true }));
    notificationHelper.putNotificationsForCurrentUser(seenNotifications);
  };
  const storageHandler = () => {
    readCurrentNotifications(setNotifications);
  };
  useEffect(() => {
    readCurrentNotifications(setNotifications);
    document.addEventListener(KEY_EVENT_INSERTED, storageHandler, false);
    return () => {
      document.removeEventListener(KEY_EVENT_INSERTED, storageHandler);
    };
  }, []);

  const closeModal = () => {
    setModalVisible(false);
  };

  const toggleModal = () => {
    setModalVisible(!modalVisible);
  };

  const { onLogout, onChangeLanguage, language } = props;
  return (
    <React.Fragment>
      <AppSidebarToggler className="d-lg-none" display="md" mobile />
      <AppNavbarBrand
        full={{
          src: logo,
          width: 45,
          height: 45,
          alt: 'Seller Center Logo',
        }}
        minimized={{
          src: logo,
          width: 35,
          height: 35,
          alt: 'Seller Center Logo',
        }}
      />
      <AppSidebarToggler className="d-md-down-none" display="lg" />

      <Nav className="d-md-down-none" navbar>
        <NavItem className="px-3">
          <NavLink href="/">Seller Center</NavLink>
        </NavItem>
      </Nav>
      <Nav className="ml-auto" navbar>
        <AppHeaderDropdown direction="down">
          <DropdownToggle nav>
            <Region language={language} />
          </DropdownToggle>
          <DropdownMenu>
            {Object.keys(regions).map(el => (
              <DropdownItem key={el} onClick={() => onChangeLanguage(el)}>
                <Region language={el} />
              </DropdownItem>
            ))}
          </DropdownMenu>
        </AppHeaderDropdown>
        <AppHeaderDropdown direction="down">
          <DropdownToggle nav>
            <i className="icon-bell" />
            {totalUnSeenNotification > 0 && (
              <Badge pill color="danger">
                {totalUnSeenNotification}
              </Badge>
            )}
          </DropdownToggle>
          <DropdownMenu right style={dropDownStyle} className="text-wrap">
            <DropdownItem header tag="div" className="d-flex">
              <strong>Thông báo</strong>
              {/* eslint-disable-next-line jsx-a11y/click-events-have-key-events */}
              <strong
                tabIndex={0}
                data-testid="seen-all"
                className="ml-auto clickable px-1"
                role="button"
                onClick={seenAllNotifications}
              >
                Đánh dấu đã đọc
              </strong>
            </DropdownItem>
            {sortedNotifications
              .slice(0, 5)
              .map(({ title, message, type, createdAt, id, seen }) => {
                return (
                  <DropdownItem key={id} className="link">
                    <div className="d-flex align-items-center">
                      <i className={`fa fa-bell-o text-${type}`} />
                      <b
                        className={
                          seen ? 'text-secondary text-wrap' : 'text-wrap'
                        }
                      >
                        {title}
                      </b>
                      <small className="ml-auto pl-5">
                        {dateTimeHelper.formatTime(moment(createdAt))}
                      </small>
                    </div>
                    <div>
                      <small className="text-wrap">{message}</small>
                    </div>
                  </DropdownItem>
                );
              })}
            <DropdownItem
              className="text-center clickable"
              data-testid="view-all"
              onClick={toggleModal}
            >
              <b>View All</b>
            </DropdownItem>
          </DropdownMenu>
        </AppHeaderDropdown>
        <AppHeaderDropdown direction="down">
          <DropdownToggle nav>
            <span>{`Hi, ${currentUser && currentUser.name}`}</span>
            <img
              src={
                (currentUser && currentUser.avatarUrl) ||
                'https://s3.amazonaws.com/uifaces/faces/twitter/alexivanichkin/128.jpg'
              }
              className="img-avatar"
              alt="User"
            />
          </DropdownToggle>
          <DropdownMenu right style={{ right: 'auto' }}>
            <DropdownItem id="log-out-btn" onClick={e => onLogout(e)}>
              <i className="fa fa-lock" /> Đăng xuất
            </DropdownItem>
          </DropdownMenu>
        </AppHeaderDropdown>
      </Nav>
      {modalVisible && (
        <NotificationModal
          visible={modalVisible}
          onCancel={closeModal}
          notifications={sortedNotifications}
        />
      )}
    </React.Fragment>
  );
};
