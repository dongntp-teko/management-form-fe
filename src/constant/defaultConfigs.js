import { paginationConstants } from 'constant/index';

const antTable = {
  locale: {
    emptyText: 'Không có sản phẩm nào',
  },
  bordered: true,
  size: 'medium',
  pagination: {
    pageSizeOptions: paginationConstants.pageSizes,
    showSizeChanger: true,
    size: 'medium',
  },
};

export default {
  antTable,
};
