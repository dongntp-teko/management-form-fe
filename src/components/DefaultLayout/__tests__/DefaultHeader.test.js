import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import { MemoryRouter } from 'react-router-dom';
import { fireEvent, render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import { setGlobal } from 'reactn';
import { languageConstants, localStorageConstants } from 'constant';
import i18n from 'i18n';
import DefaultHeader from '../DefaultHeader';

i18n.changeLanguage(languageConstants.VIETNAM);

const props = {
  onLogout: () => {},
};

setGlobal({
  [localStorageConstants.NOTIFICATION_KEY]: [
    {
      id: 1,
      title: 'hello',
      message: 'hello',
      createdAt: new Date().toISOString(),
      type: 'success',
    },
  ],
});

describe('test DefaultHeader component', () => {
  it('should render correctly', () => {
    const { getByTestId, container } = render(<DefaultHeader language="vi" />);
    fireEvent.click(getByTestId('seen-all'));
    fireEvent.click(getByTestId('view-all'));
    expect(container.firstChild).toMatchSnapshot();
  });
});

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <DefaultHeader language="vi" />
    </MemoryRouter>,
    div,
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('test click logout', () => {
  const wrapper = mount(
    <MemoryRouter>
      <DefaultHeader language="vi" {...props} />
    </MemoryRouter>,
  );
  wrapper.find('button#log-out-btn').simulate('click');
});
