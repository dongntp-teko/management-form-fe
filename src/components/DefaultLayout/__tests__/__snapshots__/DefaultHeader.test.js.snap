// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DefaultHeader component should render correctly 1`] = `
<button
  class="d-lg-none navbar-toggler"
  data-sidebar-toggler="true"
  type="button"
>
  <span
    class="navbar-toggler-icon"
  />
</button>
`;
